### PR TITLE
Exe do not inclue app package

### PR DESCRIPTION
**Pull request by wli3**
…soft.NETCore.App

Fix https://github.com/dotnet/sdk/issues/2204#issuecomment-404409125

### CODE REVIEW

File: src/Tests/Microsoft.NET.Pack.Tests/GivenThatThereAreImplicitPackageReferences.cs [hunk 1]
```diff
@@ -155,6 +155,24 @@ public void Packing_a_netcoreapp_2_0_app_includes_the_implicit_dependency()
 
             var dependencies = PackAndGetDependencies(testProject);
 
+            dependencies.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void Packing_a_netcoreapp_2_0_DotnetCliTool_app_does_not_include_the_implicit_dependency()
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Pack.Tests/GivenThatThereAreImplicitPackageReferences.cs]: Remove the "_not_" from this test name, it's testing the opposite.

**Review by dsplaisted — APPROVED** [review]

**Comment by dsplaisted** [comment]
@Pilchie Do we need any sort of approval to check in to 2.2.1xx?  Also FYI this is a breaking change, but we think the impact is low, there is a workaround, and it is worth doing because it fixes a lot of issues.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props [hunk 2]
```diff
@@ -48,10 +48,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
     <PackageReference Include="Microsoft.NETCore.App" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
 
-    <!-- For libraries targeting .NET Core 2.0 or higher, don't include a dependency on Microsoft.NETCore.App in the package produced by pack.
-         Packing an app (for example a .NET CLI tool) should include the Microsoft.NETCore.App package dependency. -->
+    <!-- For targeting .NET Core 2.0 or higher, don't include a dependency on Microsoft.NETCore.App in the package produced by pack.
+         Packing an DotnetCliTool should include the Microsoft.NETCore.App package dependency. -->
     <PackageReference Update="Microsoft.NETCore.App"
-                      Condition="('$(OutputType)' != 'Exe') And ('$(_TargetFrameworkVersionWithoutV)' != '') And ('$(_TargetFrameworkVersionWithoutV)' >= '2.0')"
+                      Condition="('$(_TargetFrameworkVersionWithoutV)' != '') And ('$(_TargetFrameworkVersionWithoutV)' >= '2.0') And ('$(PackageType)' != 'DotnetCliTool')"
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Do we want another escape hatch than setting PackageType? What else does that impact?

**Review by nguerrera — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props (hunk 2 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: I would say we already have another escape hatch: put the following in your project:

```xml
<ItemGroup>
    <PackageReference Update="Microsoft.NETCore.App" PrivateAssets="None" />
</ItemGroup>
```

**Comment by Pilchie** [comment]
I think 2.2.1xx is fair game for now (with signoff, etc)

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props (hunk 2 above)
Comment (nguerrera) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props]: Good point. I'm convinced. :)

**Comment by wli3** [comment]
Test failed due to project tools test assert don't have PackageType property https://github.com/dotnet/sdk/blob/0af8aa984773c32752634a62c87edcceedcb93e3/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateADepsFileForATool.cs#L36

I believe to be valid project tool, you need to set that package type. So I added it to the test asset

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT Release

**Pull request merged**